File: src/types/session.ts
export interface Session {
  id: string
  projectId: string
  title: string
  description: string
  status: 'active' | 'completed' | 'archived'
  createdAt: string
  updatedAt: string
  completedAt: string | null
  // Computed fields from database JOINs
  issueCount?: number
  captureCount?: number
}

export interface CreateSessionPayload {
  projectId: string
  title: string
  description?: string
}

export interface UpdateSessionPayload {
  title?: string
  description?: string
  status?: 'active' | 'completed' | 'archived'
}

File: src-tauri/src/models/session.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Session {
    pub id: String,
    pub project_id: String,
    pub title: String,
    pub description: String,
    pub status: String, // 'active' | 'completed' | 'archived'
    pub created_at: String,
    pub updated_at: String,
    pub completed_at: Option<String>,
    
    // Computed fields from database JOIN queries
    pub issue_count: Option<i64>,
    pub capture_count: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateSessionPayload {
    pub project_id: String,
    pub title: String,
    pub description: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateSessionPayload {
    pub title: Option<String>,
    pub description: Option<String>,
    pub status: Option<String>,
}
